codec/test_codec.go

```diff
@@ -86,6 +88,12 @@ type MyInnerStruct3 struct {
 	F   Foo           `serialize:"true"`
 }
 
+type MyStructWithPtr struct {
```

Comment:
I feel like this struct (and the below test) should be referencing `nullable`. Perhaps `MyStructWithNullablePtr`?